google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/StartBatchDdl.java

```diff
@@ -40,7 +47,11 @@ public boolean appliesToDialect(Dialect dialect) {
 
   @Override
   public boolean beforeTablesOnCreation() {
-    return true;
+    if (schemaAction == Action.UPDATE) {
```

Comment:
I see this logic repeated in several places. Can we refactor to have it in one place?

Reply:
Done.